website/map/views.py

```diff
@@ -232,6 +233,19 @@ def index():
     if current_user.is_anonymous:
         return redirect("/about")
 
+    try:
+        hostname, _, _ = socket.gethostbyaddr(request.remote_addr)
+
+        suffix = ".inf.ed.ac.uk"
+        if hostname.endswith(suffix):
+            machine = hostname[:-len(suffix)]
+            if machine.isalpha() and (flask_redis.exists(machine) == 1):
+                r = flask_redis.hget(machine, "room")
+                return redirect("/site/"+r)
+
+    except socket.herror:
```

Comment:
What can cause this error?

Reply:
unknown hosts

https://docs.python.org/2/library/socket.html#socket.herror

Reply:
`s/2/3/`

Reply:
same